### PR TITLE
Easier output creation with byte cost as amount

### LINKED ISSUE #1268: Easier output creation with minimum storage deposit
## Description

Make it possible to create outputs with the minimum required storage deposit without calculating it with the same output before
Some `.use_minimum_storage_deposit(byte_cost_config)` method (name can be different) in the output builder would probably be the easiest way

## Motivation

Easier to use, now I use https://github.com/iotaledger/wallet.rs/blob/shimmer-develop/src/account/operations/transfer/high_level/minimum_storage_deposit.rs only for that

## Requirements

1. Make it easier than now, so https://github.com/iotaledger/wallet.rs/blob/shimmer-develop/src/account/operations/transfer/high_level/minimum_storage_deposit.rs isn't needed anymore

## Open questions (optional)

Instead of `OutputBuilder::new(amount).use_minimum_storage_deposit()`would `Output::new_with_minimum_storage_deposit(byte_cost_config)` the cleaner way, because one doesn't has to provide some random `amount` first?

## Are you planning to do it yourself in a pull request?

No.

### PR DESCRIPTION
**Pull request by thibault-martinez**
## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

File: bee-message/src/output/alias.rs [hunk 1]
```diff
@@ -170,7 +192,18 @@ impl AliasOutputBuilder {
...
-        })
+        };
+
+        output.amount = match (self.amount, self.byte_cost_config) {
+            (Some(amount), None) => amount,
+            (None, Some(byte_cost_config)) => Output::Alias(output.clone())
+                .byte_cost(&byte_cost_config)
+                .try_into()
+                .map_err(Error::InvalidOutputAmount)?,
+            _ => unreachable!(),
```
Comment (pvdrz) [inline bee-message/src/output/alias.rs]: This makes me thing that `amount` and `byte_cost_config` should be something like `Result<OutputAmount, ByteCostConfig>`

**Review by pvdrz — CHANGES_REQUESTED** [review]

File: bee-message/src/output/alias.rs (hunk 1 above)
Comment (thibault-martinez) [reply bee-message/src/output/alias.rs]: I proposed something very similar to @Thoralf-M but it looked like it was better in terms of API to have 2 different methods.

Comment (thibault-martinez) [reply bee-message/src/output/alias.rs]: I mean, we could actually do a mix of both. Have the two methods AND the enum.

Comment (pvdrz) [reply bee-message/src/output/alias.rs]: Yeah I'd keep the methods but use an enum internally so these illegal states cannot be represented at all.

**Review by pvdrz — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**